test/test_misc.py

```diff
@@ -1844,6 +1844,28 @@ def forward(self, x):
         self.assertEqual(y, 11)
         self.assertEqual(z, 61)
 
+    def test_slicing_dynamic_shape(self):
+        def fn(y):
+            x = torch.ones(8)
+            idx = y[0]
+            out = x[idx:]
+            return (out + 3) * 5
+
+        counter = torchdynamo.testing.CompileCounter()
+        opt_fn = torchdynamo.optimize(counter)(fn)
+        out = opt_fn(torch.ones(10, dtype=torch.long))
+        # idx should be 1 -> slicing off [1:] of 8 elem tensor
+        self.assertEqual(list(out.shape), [7])
+
+        dyn = torchdynamo.config.dynamic_shapes
```

Comment:
We have an util function [```ifdyn```](https://github.com/pytorch/torchdynamo/blob/d88a0ac0e4a799617e59a05fec79d329781f3ea7/test/test_repros.py#L32) can help this.